apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1.Tests/SpannerClientImplTest.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Grpc.Core;
+using System;
+using Xunit;
+
+namespace Google.Cloud.Spanner.V1.Tests
+{
+    public class SpannerClientImplTest
+    {
+        [Fact]
+        public void ClientSetsResourcePrefixHeader()
+        {
+            var invoker = new FakeCallInvoker();
+            var client = new SpannerClientBuilder { CallInvoker = invoker }.Build();
+            const string session = "projects/proj/instances/inst/databases/db/sessions/sess";
+            client.BeginTransaction(new BeginTransactionRequest { Session = session });
+            var found = false;
+            foreach (Metadata.Entry e in invoker.Metadata)
```

Comment:
I'd suggest using:

```csharp
Metadata.Entry entry = invoker.Metadata.Single(e => e.Key == "google-cloud-resource-prefix");
Assert.Equal("projects/proj/instances/inst/databases/db", entry.Value);
```

Reply:
Hmmm I was trying to do something similar by following this [example](https://github.com/googleapis/google-cloud-dotnet/blob/d1056b9a083ab925bbe63053624b79f108d0c3ee/apis/Google.Cloud.Language.V1/Google.Cloud.Language.V1.Tests/ApiClientHeaderTest.cs#L32). Except the compiler kept telling me that `Metadata` doesn't contain a definition for `FirstOrDefault` and `Single`.

I looked at the [Metadata docs](https://grpc.github.io/grpc/csharp-dotnet/api/Grpc.Core.Metadata.html#methods) and indeed those methods are not defined. Then I thought I would use `Metadata.Contains` but xUnit was unhappy about that.

Some Googling gave me the idea to use Assert.Contains() which finally worked :)

Reply:
`Metadata` implements `IList<Metadata.Entry>`. `Single` is an extension method defined in LINQ, so `using System.Linq;` would have worked.

However, thinking about it, `Assert.Single` is *really* what you want:

```csharp
Metadata.Entry entry = Assert.Single(invoker.Metadata, e => e.Key == "google-cloud-resource-prefix");
Assert.Equal("projects/proj/instances/inst/databases/db", entry.Value);
```

That way you'll get a better error message if there are no/multiple matches.

Reply:
You're right. We don't want to set it multiple times. Updated the implementation and also added tests for all methods. There one (ListSessions) that I can't get working so I will look into it tomorrow.